### PR TITLE
Add instrumentation about notifications

### PR DESCRIPTION
**Pull request by fabxc**
@brian-brazil

### CODE REVIEW

File: notify/impl.go [hunk 1]
```diff
@@ -38,6 +39,25 @@ import (
...
+	numNotifications = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "alertmanager",
+		Name:      "notifications_total",
+		Help:      "The total number of attempted notifications.",
+	}, []string{"receiver", "integration"})
+
+	numFailedNotifications = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "alertmanager",
+		Name:      "notifications_failed_total",
+		Help:      "The total number failed notifications.",
```
Comment (brian-brazil) [inline notify/impl.go]: of

File: notify/impl.go [hunk 2]
```diff
@@ -68,14 +93,27 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 			if len(res) == 0 {
 				return nil
 			}
-			return n.Notify(ctx, res...)
+
+			var (
+				integr = n.name()
+				// The receiver is of the form <name>/<integration>/<num>.
+				// We only care about the first component.
+				recv = strings.Split(receiver(ctx), "/")[0]
```
Comment (brian-brazil) [inline notify/impl.go]: Are names limited from having slashes?

File: notify/impl.go [hunk 3]
```diff
@@ -38,6 +39,25 @@ import (
 	"github.com/prometheus/alertmanager/types"
 )
 
+var (
+	numNotifications = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "alertmanager",
+		Name:      "notifications_total",
+		Help:      "The total number of attempted notifications.",
+	}, []string{"receiver", "integration"})
```
Comment (brian-brazil) [inline notify/impl.go]: I can imagine the number of receivers getting really high (thousands) in large companies, I wonder if we should just split out by integration and leave the rest to logs.

File: notify/impl.go (hunk 2 above)
Comment (fabxc) [inline notify/impl.go]: Currently not. It's sort of dirty in general restoring the info from a string that should be opaque – I might have to think about some tweaks elsewhere to do this better.

File: notify/impl.go (hunk 3 above)
Comment (fabxc) [inline notify/impl.go]: Made it so.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**